### PR TITLE
Update metrics docs

### PR DESCRIPTION
**Pull request by dagnir**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [x] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: docs/design/core/metrics/MetricsList.md [hunk 1]
```diff
@@ -1,130 +1,61 @@
...
+| Name                          | Type          | Description |
+|-------------------------------|---------------|-------------|
+| ServiceId                     | `String`      | The unique ID for the service. This is present for all API call metrics.|
+| OperationName                 | `String`      | The name of the service operation being invoked. This is present for all API call metrics.|
+| ApiCallDuration               | `Duration`    | The duration of the API call. This includes all call attempts made.|
+| ApiCallSuccessful             | `Boolean`     | True if the API call succeeded, false otherwise. |
+| BackoffDelayDuration          | `Duration`    | The duration of time that the SDK has waited before this API call attempt, based on the retry policy. |
+| MarshallingDuration           | `Duration`    | The duration of time taken to marshall the SDK request to an HTTP request.|
+| CredentialsFetchDuration      | `Duration`    | The duration of time taken to fetch signing credentials for the request.|
+| SigningDuration               | `Duration`    | The duration fo time taken to sign the HTTP request.|
```
Comment (Quanzzzz) [inline docs/design/core/metrics/MetricsList.md:27]: Typo, "of"

File: docs/design/core/metrics/MetricsList.md [hunk 2]
```diff
@@ -1,130 +1,61 @@
...
+| HttpRequestRoundTripTime      | `Duration`    | The total time take to send a HTTP request and receive the response.|
+| AwsRequestId                  | `String`      | The extended request ID of the service request.|
```
Comment (Quanzzzz) [inline docs/design/core/metrics/MetricsList.md:29]: The description is the same as the next line, are they the same thing?

Comment (dagnir) [reply docs/design/core/metrics/MetricsList.md]: Nice catch, fixed

File: docs/design/core/metrics/MetricsList.md (hunk 1 above)
Comment (dagnir) [reply docs/design/core/metrics/MetricsList.md]: Fixed

**Review by Quanzzzz — APPROVED** [review]

**Pull request merged**